packages/builders/src/components/selectMenu/StringSelectMenu.ts

```diff
@@ -108,36 +142,44 @@ export class StringSelectMenuBuilder extends BaseSelectMenuBuilder<APIStringSele
 	public spliceOptions(
 		index: number,
 		deleteCount: number,
-		...options: RestOrArray<APISelectMenuOption | StringSelectMenuOptionBuilder>
+		...options: (
+			| APISelectMenuOption
+			| StringSelectMenuOptionBuilder
+			| ((builder: StringSelectMenuOptionBuilder) => StringSelectMenuOptionBuilder)
+		)[]
 	) {
-		const normalizedOptions = normalizeArray(options);
+		const resolved = options.map((option) => {
+			if (option instanceof StringSelectMenuOptionBuilder) {
+				return option;
+			}
 
-		const clone = [...this.options];
+			if (typeof option === 'function') {
+				return option(new StringSelectMenuOptionBuilder());
+			}
 
-		clone.splice(
-			index,
-			deleteCount,
-			...normalizedOptions.map((normalizedOption) =>
-				normalizedOption instanceof StringSelectMenuOptionBuilder
-					? normalizedOption
-					: new StringSelectMenuOptionBuilder(jsonOptionValidator.parse(normalizedOption)),
-			),
-		);
+			return new StringSelectMenuOptionBuilder(option);
+		});
+
+		this.data.options ??= [];
+		this.data.options.splice(index, deleteCount, ...resolved);
 
-		optionsLengthValidator.parse(clone.length);
-		this.options.splice(0, this.options.length, ...clone);
 		return this;
 	}
 
 	/**
-	 * {@inheritDoc BaseSelectMenuBuilder.toJSON}
+	 * {@inheritDoc ComponentBuilder.toJSON}
 	 */
-	public override toJSON(): APIStringSelectComponent {
-		validateRequiredSelectMenuParameters(this.options, this.data.custom_id);
+	public override toJSON(validationOverride?: boolean): APIStringSelectComponent {
+		const { options, ...rest } = this.data;
+		const data = {
+			...(structuredClone(rest) as APIStringSelectComponent),
+			options: options?.map((option) => option.toJSON(false)),
```

Comment:
```suggestion
			options: options?.map((option) => option.toJSON(validationOverride)),
```

Reply:
This is on purpose, if you look at the `selectMenuStringPredicate` it does this:
```ts
export const selectMenuStringPredicate = selectMenuBasePredicate.extend({
	type: z.literal(ComponentType.StringSelect),
	options: selectMenuStringOptionPredicate.array().max(25).optional(),
});
```

In other words, if we passed `validationOverride`, each option would basically be validated twice. There's places where this is incredibly hard for me to do because of different types of things (esp. slash command options) and figuring out what predicate to use. In those places, I pass `validationOverride` down instead. If we want a consistent pattern on this, I'd probably go with yours, but I'd have to adjust the predicates a bit (e.g. makae the above like, `z.any().array().max(25).optional()` and leave the individual validation to the specific toJSON method)

Reply:
Gotcha. Let's add a comment then in every place where this is done so we know it's intentional please 🙏

Reply:
```suggestion
			options: options.map((option) => option.toJSON(false)),
```